src/apps/chatter/components/chatbox/chatbox.js

```diff
@@ -1,24 +1,29 @@
-import './chatbox.less'
-import React, { useState, useEffect } from 'react'
+import React, { useState, useEffect, useRef } from 'react'
+
 import { Message } from '/elements/message'
-import { useRoom } from '../../hooks/use-room'
+import { useRoom } from '/apps/chatter/hooks/use-room'
+import './chatbox.less'
 
 export const ChatBox = ({ roomId }) => {
   const [textInput, setTextInput] = useState('')
   const [messages, sendNewMessage] = useRoom(roomId)
+  const inputEl = useRef(null)
 
   useEffect(() => {
-    document.querySelector('#chatbox__message-input').focus()
-    document.querySelector('#chatbox__message-input').onkeyup = function (e) {
-      if (e.keyCode === 13) {
-        document.querySelector('.chatbox__send-button').click()
-      }
-    }
-  }, [])
+    inputEl.current.focus()
+  }, [messages])
```

Comment:
Hmm, I feel like this could result in funny UX. If I were typing somewhere else, even the address bar, or if I'm on a screen reader, or focused on another element, I could lose my focus and not know why.

I think it makes sense to focus the input on load, and to maintain focus after hitting enter on a new message, but no more than that.

What do you think? DOes this cover another use case I'm missing?

Reply:
It was like that originally, but would only work when there are no messages. When the messages render, the input loses focus and looks like it never had it. This is the only workaround I found, any other ideas?